zerver/tests/test_email_mirror.py

```diff
@@ -54,24 +54,54 @@ def test_encode_decode(self) -> None:
         self.assertTrue(email_address.endswith('@testserver'))
         tup = decode_email_address(email_address)
         assert tup is not None
-        (decoded_stream_name, token) = tup
+        (decoded_stream_name, token, show_sender) = tup
+        self.assertFalse(show_sender)
+        self.assertEqual(decoded_stream_name, stream_name)
+        self.assertEqual(token, stream.email_token)
+
+        parts = email_address.split('@')
+        parts[0] += "+show-sender"
+        email_address_show = '@'.join(parts)
+        tup = decode_email_address(email_address_show)
+        assert tup is not None
+        (decoded_stream_name, token, show_sender) = tup
+        self.assertTrue(show_sender)
         self.assertEqual(decoded_stream_name, stream_name)
         self.assertEqual(token, stream.email_token)
 
         email_address = email_address.replace('+', '.')
```

Comment:
It might be better to have this user a new variable name rather than having all future blocks deal with a mutated `email_address`.

Reply:
(e.g. `email_address_dots`)

Reply:
Okay, changed to having separate variables for the addresses with dots.